addons/stock/models/stock_location.py

```diff
@@ -336,15 +340,20 @@ def _check_can_be_used(self, product, quantity=0, package=None, location_qty=0):
         specified."""
         self.ensure_one()
         if self.storage_category_id:
-            # check weight
-            if self.storage_category_id.max_weight < self.forecast_weight + product.weight * quantity:
-                return False
             # check if enough space
             if package and package.package_type_id:
+                # check weight
+                package_smls = self.env['stock.move.line'].search([('result_package_id', '=', package.id)])
+                if self.storage_category_id.max_weight < self.forecast_weight + sum(package_smls.mapped(lambda sml: sml.product_uom_qty * sml.product_id.weight)):
```

Comment:
I think `sml.product_qty` is better as the weight is expressed in the product’s uom